编译期报错，javassist新增方法报错，javassist.CannotCompileException: [source error] ) is missing

在source项目下，在simple-app-lib中新增了两个360的jar包，然后编译插件报错：

Execution failed for task ':sample-plugin-app:transformClassesWithShadowTransformForDebug'.
> javassist.CannotCompileException: [source error] ) is missing
* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':sample-plugin-app:transformClassesWithShadowTransformForDebug'
Caused by: javassist.CannotCompileException: [source error] ) is missing
	at javassist.CtBehavior.setBody(CtBehavior.java:474)
	at javassist.CtBehavior.setBody(CtBehavior.java:440)
	at com.tencent.shadow.core.transform.specific.PackageManagerTransform$setupPackageManagerTransform$$inlined$forEach$lambda$1.transform(PackageManagerTransform.kt:81)
	at com.tencent.shadow.core.transform_kit.AbstractTransformManager.fireAll(AbstractTransformManager.kt:41)
	at com.tencent.shadow.core.transform_kit.AbstractTransform.onTransform(AbstractTransform.kt:61)
	at com.tencent.shadow.core.transform_kit.ClassTransform.transform(ClassTransform.kt:173)
	at com.android.build.gradle.internal.pipeline.TransformTask$2.call(TransformTask.java:284)
	at com.android.build.gradle.internal.pipeline.TransformTask$2.call(TransformTask.java:247)
	at com.android.builder.profile.ThreadRecorder.record(ThreadRecorder.java:106)
	... 92 more
Caused by: compile error: ) is missing
	at javassist.compiler.Parser.parseArgumentList(Parser.java:1352)
	at javassist.compiler.Parser.parseMethodCall(Parser.java:1196)
	at javassist.compiler.Parser.parsePostfix(Parser.java:1054)
	at javassist.compiler.Parser.parseUnaryExpr(Parser.java:904)
	at javassist.compiler.Parser.parseBinaryExpr(Parser.java:794)
	at javassist.compiler.Parser.parseConditionalExpr(Parser.java:739)
	at javassist.compiler.Parser.parseExpression(Parser.java:719)
	at javassist.compiler.Parser.parseReturn(Parser.java:524)
	at javassist.compiler.Parser.parseStatement(Parser.java:287)
	at javassist.compiler.Javac.compileBody(Javac.java:219)
	at javassist.CtBehavior.setBody(CtBehavior.java:466)

对修改字节码不熟，不知道为什么会报这个错误?  猜测1：jar包中代码混淆的原因？ 猜测2：jar包中类的包名以数字开头的原因？

--- a/projects/sdk/core/transform/src/main/kotlin/com/tencent/shadow/core/transform/ShadowTransform.kt
+++ b/projects/sdk/core/transform/src/main/kotlin/com/tencent/shadow/core/transform/ShadowTransform.kt
@@ -29,6 +29,10 @@ class ShadowTransform(
         classPoolBuilder: ClassPoolBuilder,
         private val useHostContext: () -> Array<String>
 ) : AbstractTransform(project, classPoolBuilder) {
+    companion object {
+        const val SelfClassNamePlaceholder =
+            "com.tencent.shadow.core.transform.SelfClassNamePlaceholder"
+    }
 
     lateinit var _mTransformManager: TransformManager
 
@@ -38,6 +42,7 @@ class ShadowTransform(
     override fun beforeTransform(invocation: TransformInvocation) {
         super.beforeTransform(invocation)
         _mTransformManager = TransformManager(mCtClassInputMap, classPool, useHostContext)
+        classPool.makeInterface(SelfClassNamePlaceholder)
     }
 
     override fun isCacheable(): Boolean {

--- a/projects/sdk/core/transform/src/main/kotlin/com/tencent/shadow/core/transform/specific/PackageItemInfoTransform.kt
+++ b/projects/sdk/core/transform/src/main/kotlin/com/tencent/shadow/core/transform/specific/PackageItemInfoTransform.kt
@@ -18,6 +18,7 @@
 
 package com.tencent.shadow.core.transform.specific
 
+import com.tencent.shadow.core.transform.ShadowTransform.Companion.SelfClassNamePlaceholder
 import com.tencent.shadow.core.transform_kit.SpecificTransform
 import com.tencent.shadow.core.transform_kit.TransformStep
 import javassist.*
@@ -76,7 +77,7 @@ class PackageItemInfoTransform : SpecificTransform() {
                                 .append(".")
                                 .append(targetMethod.methodInfo.name)
                                 .append("(")
-                                .append(ctClass.name)
+                                .append(SelfClassNamePlaceholder)
                                 .append(".class.getClassLoader(),")
                         for (i in 1..newMethod.parameterTypes.size) {
                             if (i > 1) {
@@ -88,6 +89,7 @@ class PackageItemInfoTransform : SpecificTransform() {
 
                         newMethod.setBody(newBodyBuilder.toString())
                         ctClass.addMethod(newMethod)
+                        ctClass.replaceClassName(SelfClassNamePlaceholder,ctClass.name)
                         val codeConverter = CodeConverter()
                         codeConverter.redirectMethodCallToStatic(targetMethod, newMethod)
                         ctClass.instrument(codeConverter)

--- a/projects/sdk/core/transform/src/main/kotlin/com/tencent/shadow/core/transform/specific/PackageManagerTransform.kt
+++ b/projects/sdk/core/transform/src/main/kotlin/com/tencent/shadow/core/transform/specific/PackageManagerTransform.kt
@@ -18,6 +18,7 @@
 
 package com.tencent.shadow.core.transform.specific
 
+import com.tencent.shadow.core.transform.ShadowTransform.Companion.SelfClassNamePlaceholder
 import com.tencent.shadow.core.transform_kit.SpecificTransform
 import com.tencent.shadow.core.transform_kit.TransformStep
 import javassist.*
@@ -66,7 +67,7 @@ class PackageManagerTransform : SpecificTransform() {
                                 .append(".")
                                 .append(targetMethod.methodInfo.name)
                                 .append("(")
-                                .append(ctClass.name)
+                                .append(SelfClassNamePlaceholder)
                                 .append(".class.getClassLoader(),")
                         //下面放弃第0个和第1个参数，第0个是this，
                         //第1个是redirectMethodCallToStaticMethodCall时原本被调用的PackageManager对象。
@@ -80,6 +81,7 @@ class PackageManagerTransform : SpecificTransform() {
 
                         newMethod.setBody(newBodyBuilder.toString())
                         ctClass.addMethod(newMethod)
+                        ctClass.replaceClassName(SelfClassNamePlaceholder, ctClass.name)
                         val codeConverter = CodeConverter()
                         codeConverter.redirectMethodCallToStatic(targetMethod, newMethod)
                         ctClass.instrument(codeConverter)

--- a/projects/sdk/core/transform/src/test/kotlin/com/tencent/shadow/core/transform/specific/PackageManagerTransformTest.kt
+++ b/projects/sdk/core/transform/src/test/kotlin/com/tencent/shadow/core/transform/specific/PackageManagerTransformTest.kt
@@ -18,6 +18,7 @@
 
 package com.tencent.shadow.core.transform.specific
 
+import com.tencent.shadow.core.transform.ShadowTransform.Companion.SelfClassNamePlaceholder
 import com.tencent.shadow.core.transform_kit.AbstractTransformTest
 import javassist.CtClass
 import javassist.CtMethod
@@ -34,12 +35,16 @@ class PackageManagerTransformTest : AbstractTransformTest() {
 
         val packageManagerTransform = PackageManagerTransform()
         packageManagerTransform.mClassPool = sLoader
+        packageManagerTransform.mClassPool.makeInterface(SelfClassNamePlaceholder)
         packageManagerTransform.setup(allInputClass)
 
         val methods = arrayOf("getApplicationInfo","getActivityInfo")
 
 
         allInputClass.forEach {
+            //将测试类的包名改为Java的非法包名字符，测试Proguard混淆成这种形式的jar的场景
+            it.name = "1.${it.simpleName}"
+
             for (method in methods) {
                 beforeTransformCheck(it, method)
             }

